GDExtension refactor: `PropertyInfo` code reuse

Bromeon opened the pull request:
Small refactor which adds a constructor
```cpp
explicit PropertyInfo(const GDNativePropertyInfo &pinfo)
```
and uses it in several places, where we need to convert the C struct to the C++ type.

In `core/object/script_language_extension.h`, the previous code passed twice `class_name`, but never `name`. The semantics change with this PR. Is my assumption that this was a mistake correct?

Two style-related points:
* I wasn't entirely sure if the constructor should take its argument by const-pointer or const-reference, as I found both styles in Godot. Let me know if you prefer taking a pointer, I'll gladly change it.
* A delegating constructor could be used instead of an initialization list, if you prefer.

Bromeon committed:
GDExtension: reuse code with constructor PropertyInfo(const GDNativePropertyInfo&)

holgac commented on core/object/script_language_extension.h:
```diff
@@ -671,7 +671,7 @@ class ScriptInstanceExtension : public ScriptInstance {
 			const GDNativePropertyInfo *pinfo = native_info->get_property_list_func(instance, &pcount);
 			for (uint32_t i = 0; i < pcount; i++) {
-				p_list->push_back(PropertyInfo(Variant::Type(pinfo[i].type), pinfo[i].class_name, PropertyHint(pinfo[i].hint), pinfo[i].hint_string, pinfo[i].usage, pinfo[i].class_name));
+				p_list->push_back(PropertyInfo(*pinfo));
```
same here: &pinfo[i]

holgac commented on core/extension/native_extension.cpp:
```diff
@@ -204,16 +197,11 @@ void NativeExtension::_register_extension_class_property(const GDNativeExtension
 
 	StringName class_name = p_class_name;
-	ERR_FAIL_COND_MSG(!self->extension_classes.has(class_name), "Attempt to register extension class property '" + String(p_info->name) + "' for unexisting class '" + class_name + "'.");
+	String property_name = *(String *)p_info->name;
```
This cast seems wrong: p_info->name is a const char*. You can just do

String property_name(p_info->name);

Bromeon commented:
Thanks a lot for the review!
Man, those were really stupid mistakes. I should probably check my code the next day when working late 😬 

I fixed it and also rebased it on latest `master`.

akien-mga commented on core/object/object.cpp:
```diff
@@ -493,7 +493,7 @@ void Object::get_property_list(List<PropertyInfo> *p_list, bool p_reversed) cons
 		const GDNativePropertyInfo *pinfo = _extension->get_property_list(_extension_instance, &pcount);
 		for (uint32_t i = 0; i < pcount; i++) {
-			p_list->push_back(PropertyInfo(Variant::Type(pinfo[i].type), pinfo[i].class_name, PropertyHint(pinfo[i].hint), pinfo[i].hint_string, pinfo[i].usage, pinfo[i].class_name));
+			p_list->push_back(PropertyInfo(pinfo[i]));
```
There's a logic difference here (and likewise in `script_language_extension.cpp`'s replacements) if `pinfo[i].name` is not the same as `pinfo[i].class_name`. In all three cases the multi-arg constructor was assigning `class_name` to the new PropertyInfo's `name`.

akien-mga replied:
Ah I see you pointed this out in the PR description. @reduz should likely check if this was intentional or indeed an oversight.

akien-mga commented:
Thanks!
